src/cheri_insts_cext.sail

```diff
@@ -0,0 +1,170 @@
+mapping clause assembly = C_CLCSP(uimm, cd)
+  <-> "c.clcsp" ^ spc() ^ cap_reg_name(cd) ^ sep() ^ hex_bits_10(uimm)
+
+function clause execute (C_CLCSP(uimm, cd)) = {
+  execute(LoadCapImm(cd, sp, EXTZ(uimm)))
+}
+
+union clause ast = C_CSC : (bits(9), cregidx, cregidx)
+
+mapping clause encdec_compressed_capmode = C_CSC(ui8 @ ui76 @ ui54 @ 0b0000, csc1, csc2) if sizeof(xlen) == 64
+  <-> 0b101 @ ui54 : bits(2) @ ui8 : bits(1) @ csc1 : bits(3) @ ui76 : bits(2) @ csc2 : bits(3) @ 0b00 if sizeof(xlen) == 64
+mapping clause encdec_compressed_capmode = C_CSC(0b0 @ ui76 @ ui53 @ 0b000, csc1, csc2) if sizeof(xlen) == 32
+  <-> 0b111 @ ui53 : bits(3) @ csc1 : bits(3) @ ui76 : bits(2) @ csc2 : bits(3) @ 0b00 if sizeof(xlen) == 32
+mapping clause assembly = C_CSC(uimm, csc1, csc2)
+  <-> "c.csc" ^ spc() ^ cap_creg_name(csc1) ^ sep() ^ cap_creg_name(csc2) ^ sep() ^ hex_bits_9(uimm)
+
+function clause execute (C_CSC(uimm, csc1, csc2)) = {
+  execute(StoreCapImm(creg2reg_idx(csc2), creg2reg_idx(csc1), EXTZ(uimm)))
+}
+
+union clause ast = C_CSCSP : (bits(10), regidx)
+
+mapping clause encdec_compressed_capmode = C_CSCSP(ui96 @ ui54 @ 0b0000, cs2) if sizeof(xlen) == 64
+  <-> 0b101 @ ui54 : bits(2) @ ui96 : bits(4) @ cs2 : regidx @ 0b10 if sizeof(xlen) == 64
+mapping clause encdec_compressed_capmode = C_CSCSP(0b0 @ ui86 @ ui53 @ 0b000, cs2) if sizeof(xlen) == 32
+  <-> 0b111 @ ui53 : bits(3) @ ui86 : bits(3) @ cs2 : regidx @ 0b10 if sizeof(xlen) == 32
+mapping clause assembly = C_CSCSP(uimm, rs2)
+  <-> "c.cscsp" ^ spc() ^ cap_reg_name(rs2) ^ sep() ^ hex_bits_10(uimm)
+
+function clause execute (C_CSCSP(uimm, cs2)) = {
+  execute(StoreCapImm(cs2, sp, EXTZ(uimm)))
+}
+
+/* Capability arithmetic instructions */
+union clause ast = C_CIncOffset16CSP : (bits(10))
+
+mapping clause encdec_compressed_capmode = C_CIncOffset16CSP(nzi9 @ nzi87 @ nzi6 @ nzi5 @ nzi4 @ 0b0000)
+      if nzi9 @ nzi87 @ nzi6 @ nzi5 @ nzi4 != 0b000000
+  <-> 0b011 @ nzi9 : bits(1) @ /* x2 */ 0b00010 @ nzi4 : bits(1) @ nzi6 : bits(1) @ nzi87 : bits(2) @ nzi5 : bits(1) @ 0b01
+      if nzi9 @ nzi87 @ nzi6 @ nzi5 @ nzi4 != 0b000000
+mapping clause assembly = C_CIncOffset16CSP(nzimm) if nzimm != 0b0000000000
+  <-> "c.cincoffset16csp" ^ spc() ^ hex_bits_10(nzimm) if nzimm != 0b0000000000
+
+function clause execute (C_CIncOffset16CSP(nzimm)) = {
+  execute(CIncOffsetImmediate(sp, sp, EXTS(nzimm)))
+}
+
+union clause ast = C_CIncOffset4CSPN : (cregidx, bits(10))
+
+mapping clause encdec_compressed_capmode = C_CIncOffset4CSPN(cdc, nz96 @ nz54 @ nz3 @ nz2 @ 0b00)
+      if nz96 @ nz54 @ nz3 @ nz2 != 0b00000000
+  <-> 0b000 @ nz54 : bits(2) @ nz96 : bits(4) @ nz2 : bits(1) @ nz3 : bits(1) @ cdc : cregidx @ 0b00
+      if nz96 @ nz54 @ nz3 @ nz2 != 0b00000000
+mapping clause assembly = C_CIncOffset4CSPN(cdc, nzuimm) if nzuimm != 0b0000000000
+  <-> "c.cinoffet4cspn" ^ spc() ^ cap_creg_name(cdc) ^ sep() ^ hex_bits_10(nzuimm) if nzuimm != 0b0000000000
```

Comment:
```suggestion
  <-> "c.cincoffet4cspn" ^ spc() ^ cap_creg_name(cdc) ^ sep() ^ hex_bits_10(nzuimm) if nzuimm != 0b0000000000
```

Reply:
There was a missing 'c' and 's' in 'incoffset'. Not sure if the 'n' is significant?

Reply:
The n matches `c.addi4spn`